src/Slices/InstanceComposerEditor/Core/helper.ts

```diff
@@ -0,0 +1,26 @@
+import { ServiceModel } from "@/Core";
+
+/**
+ * Recursively filters an array of ServiceModel objects based on possible ownership of the root instance.
+ *
+ * @param {ServiceModel[]} services - An array of ServiceModel objects to filter.
+ * @param {ServiceModel} mainService - root service to filter owned services by it.
+ * @returns {ServiceModel[]} An array of owned ServiceModel objects
+ */
+export const filterServices = (
+  services: ServiceModel[],
+  mainService: ServiceModel,
+): ServiceModel[] => {
+  const relatedServices = services.filter((service) =>
+    mainService?.owned_entities.includes(service.name),
```

Comment:
If `mainService`  doesn't have the property, the expression` mainService?.owned_entities` will also evaluate to undefined, and trying to call `includes` on undefined will result in a runtime error.

`Array.isArray(mainService.owned_entities) && mainService.owned_entities.includes(service.name)` Could be a more defensive alternative.